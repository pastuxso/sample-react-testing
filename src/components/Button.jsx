import React from 'react';
import PropTypes from 'prop-types';

const Button = ({text, onClick}) => {
  const handleOnClick = () => {
    onClick();
  }

  return (
    <button type="button" onClick={handleOnClick}>{text}</button>
  );
}

Button.propTypes = {
  text: PropTypes.string,
  onClick: PropTypes.func,
}

Button.defaultProps = {
  text: null,
  onClick: null
}

export default Button;
