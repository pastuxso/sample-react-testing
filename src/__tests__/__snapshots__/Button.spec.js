import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from '../../components/Button';

describe('<Button />', () => {
  describe('render', () =>  {
    it('match snapshot', () => {
      const tree = renderer.create(<Button />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('match snapshot with text prop', () => {
      const tree = renderer.create(<Button text="Button text"/>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('match snapshot with onClick prop', () => {
      const mockFn = jest.fn();
      const tree = renderer.create(<Button onClick={mockFn} text="text"/>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('#onClick', () => {
    it ('the callback should be called', () => {
      const text = "Click me please!"
      const mockFn = jest.fn();

      render(<Button onClick={mockFn} text={text}/>);

      fireEvent.click(screen.getByText(text));

      expect(mockFn).toHaveBeenCalled();
    })
  });
});
