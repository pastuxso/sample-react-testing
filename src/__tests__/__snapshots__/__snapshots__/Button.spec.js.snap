// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render match snapshot 1`] = `
<button
  onClick={[Function]}
  type="button"
/>
`;

exports[`<Button /> render match snapshot with onClick prop 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  text
</button>
`;

exports[`<Button /> render match snapshot with text prop 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Button text
</button>
`;
